edb/schema/links.py

```diff
@@ -317,10 +317,10 @@ def _apply_field_ast(
                         node.target = expr.qlast
                     else:
                         t = op.new_value
-                        assert isinstance(t, so.Object)
+                        assert isinstance(t, (so.Object, so.ObjectShell))
```

Comment:
In places like this, don't you want to explicitly resolve `ObjectShell` into an `Object` and then pass it down?

Reply:
Not really, because the delta you're decompiling to DDL might not have actually been applied, so the object isn't necessarily in the schema.